### PR TITLE
feat: add support for import-alias-naming rule

### LINKED ISSUE #880: Linter rule to check against import alias naming restrictions
**Is your feature request related to a problem? Please describe.**

Go gives some recommendations on [how packages should be named](https://go.dev/blog/package-names). This should also imply the rules for the aliases. I'd like to check that aliases for packages in our project don't include CAPS and underscores.

**Describe the solution you'd like**
I have two options:

* Regex mask that would fit any possible extra limitations that the project might want to have (for example completely forbid `v1` alias)
* Add 2 configuration options: `forbidCAPS` and `forbidUnderscores`.

The first one is more universal, the second one is simpler and more precise. We can also combine both options but then it might bring confusion.

NB: in any case we'll skip `_` import alias used for side effects.

**Describe alternatives you've considered**
I checked `importas` linter and it didn't have this feature. I submitted a feature request there, but I'm not sure if it fits the idea of `importas`.

**Additional context**
Example:

```go
package foo

import (
    fooBAR "foo/bar" // MATCH /import alias should not include capital letters/
    bar_foo "bar/foo" // MATCH /import alias should not include underscores/
    _ "far/boo" // this is fine
)
```

OR

```go
package foo

import (
    foo_BAR "foo/bar" // MATCH /import alias must much the regular expression: "^[a-z][a-z0-9]{0,}$"/
    _ "far/boo" // this is fine
)
```

### PR DESCRIPTION
**Pull request by denisvmedia**
Closes #880.

### CODE REVIEW

**Comment by denisvmedia** [comment]
@chavacava I'm thinking of making it an array of rules instead of a single rule. This will let users submitting multiple regexes that will be combined as logical AND (so that every rule must match). What do you think?

File: rule/import-naming.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+// ImportNamingRule lints import naming.
+type ImportNamingRule struct {
+	configured       bool
+	namingRule       string
+	namingRuleRegexp *regexp.Regexp
+	sync.Mutex
+}
+
+const defaultNamingRule = "^[a-z][a-z0-9]$"
```
Comment (chavacava) [inline rule/import-naming.go:19]:
```suggestion
const defaultNamingRule = "^[a-z][a-z0-9]*$"
```

**Review by chavacava — COMMENTED** [review]
Thanks for the PR! I've left some comments.

**Comment by chavacava** [comment]
> @chavacava I'm thinking of making it an array of rules instead of a single rule. This will let users submitting multiple regexes that will be combined as logical AND (so that every rule must match). What do you think?
> upd: to be clear, the idea is clarity of the configuration. The first regex can be the main, positive, match. Other subsequent regexes may be used to imply negative matches. E.g. to exclude `v1` the user will provide a negative match `^(?!v1$).*`. However, it is possible to do in a single regex, the readability will suffer, I'm afraid.

I personally prefer a single argument. That way there is no need to explain (and understand) how multiple regex are combined (AND or OR?). Yes it can lead to a complex regex but... if you are playing with a regex you know what are you doing :)

File: rule/import-naming.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	"regexp"
+	"sync"
+
+	"github.com/mgechev/revive/lint"
+)
+
+// ImportNamingRule lints import naming.
+type ImportNamingRule struct {
+	configured       bool
+	namingRule       string
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply rule/import-naming.go:14]: I used it in the error reporting. But we can use the namingRuleRegexp.String() instead. So, I will remove it.

File: rule/import-naming.go (hunk 1 above)
Comment (denisvmedia) [reply rule/import-naming.go:19]: It was a mistake. It should have been:

"^[a-z][a-z0-9]{0,}$"

Adding `*` will mean only one symbol. I will update it.

File: rule/import-naming.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+		alias := is.Name
+		if alias == nil || alias.Name == "_" {
+			continue
+		}
+
+		if !r.namingRuleRegexp.MatchString(alias.Name) {
+			failures = append(failures, lint.Failure{
+				Confidence: 1,
+				Failure:    fmt.Sprintf("import name (%s) must match the regular expression: %s", alias.Name, r.namingRule),
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply rule/import-naming.go:69]: Agree. I will update it.

File: RULES_DESCRIPTIONS.md [hunk 4]
```diff
@@ -489,6 +490,22 @@ name of an imported package. This rule spots identifiers that shadow an import.
 
 _Configuration_: N/A
 
+## import-naming
```
[the remark this answers is not in the document]
Comment (denisvmedia) [reply RULES_DESCRIPTIONS.md]: Done.

File: rule/import-naming.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (denisvmedia) [reply rule/import-naming.go]: Done.

File: rule/import-naming.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (denisvmedia) [reply rule/import-naming.go]: Done.

File: rule/import-naming.go (hunk 1 above)
Comment (denisvmedia) [reply rule/import-naming.go]: Applied according to my pattern.

**Comment by denisvmedia** [comment]
> I personally prefer a single argument. That way there is no need to explain (and understand) how multiple regex are combined (AND or OR?). Yes it can lead to a complex regex but... if you are playing with a regex you know what are you doing :)

OK, let it be a single regex.

**Pull request merged**